src/Native/Windows/build-native.cmd

```diff
@@ -10,36 +10,53 @@ set __IntermediatesDir=""
 set __BuildArch=AnyCPU
 set __VCBuildArch=x86_amd64
 set CMAKE_BUILD_TYPE=Release
+set __EnvironmentSet=false
+set __VSVersion=vs2013
```

Comment:
Some general background here to frame the discussion: CMake requires the use of a "generator" to produce the intermediary files that can then be compiled to a dll or lib file. On Unix, the generator is Clang and it produces a MakeFile. On Windows, the generator is Visual Studio and it produces a vcxproj.

The Unix and Windows native builds therefore have two steps:
- Call CMake on the sources to produce the intermediate files using clang (unix) or VS (windows) as the generator
- Call Make (unix) or msbuild (Windows) on the intermediate files to produce the native binary

>we should use CMake the same way on windows and linux/osx

The goal I'm moving towards is to enable cross-compilation of CMake such that the native binaries for Unix may be built on Windows and vice-versa, but building them the <b>same</b> way is a bit more complicated. The main issue blocking this is the lack of a stable generator that is compatible with both platforms; clang for Windows is still early stages, and msbuild on Unix may not adequately replace full VS enough to qualify as a generator. I really need to delve deeper into Clang for Windows since that's likely our best option for building the same way, but I haven't a chance to do so yet.

If we just wanted to produce the same <i>type</i> of intermediate file then we could use nmake as the generator instead of VS for Windows, but that still isn't the same method as Unix since the generator would be different.

>but all I see is that we have a bunch of extra scripts specific to windows which I would like to eliminate if possible.

That's my eventual goal after cross-compilation support. I'd like to have the logic for the Unix build the same as the Windows logic, generator and all, but it will take some extra work that is mostly orthogonal to getting the internal build and open build at parity.

>Also at one point we talked about having cmake stop generating VS solutions for windows and have it consistent for both windows and x-plat and just using make. Is that something you have looked into yet?

In short: yes I have looked into it, but not enough to be ready to move forward with.